www/js/lib/uiUtil.js

```diff
@@ -1003,6 +1011,436 @@ function warnAndOpenExternalLinkInNewTab (event, clickedAnchor, archive) {
+                        const images = articleBody.querySelectorAll('img');
+                        let firstImage = null;
+                        if (images && params.contentInjectionMode === 'serviceworker') {
+                            // Iterate over images until we find one with a width greater than 50 pixels
+                            // (this filters out small icons)
+                            const imageArray = Array.from(images);
+                            for (let j = 0; j < imageArray.length; j++) {
+                                if (imageArray[j] && imageArray[j].width > 50) {
+                                    firstImage = imageArray[j];
+                                    break;
+                                }
+                            }
+                        }
+                        if (firstImage) {
+                            // Calculate root relative URL of image
+                            const imageZimURL = encodeURI(deriveZimUrlFromRelativeUrl(firstImage.getAttribute('src'), balloonBaseURL));
+                            firstImage.src = rootRelativePathPrefix + imageZimURL;
+                            balloonString = firstImage.outerHTML + balloonString;
+                        }
+                        if (!balloonString) {
+                            reject(new Error('No article lede or image'));
+                        } else {
+                            resolve(balloonString);
+                        }
+                    } else {
+                        reject(new Error('No article body found'));
+                    }
+                });
+            });
+        }
+        if (!dirEntry) {
+            return Promise.reject(new Error('No directory entry found'));
+        } else if (dirEntry.redirect) {
+            // If the dirEntry is a redirect, we need to resolve it before reading the article
+            return new Promise((resolve, reject) => {
+                archive.resolveRedirect(dirEntry, function (reDirEntry) {
+                    resolve(readArticle(reDirEntry));
+                });
+            }).catch(error => {
+                return Promise.reject(error);
+            });
+        } else {
+            // Directory entry was found, so now read the article data
+            return Promise.resolve(readArticle(dirEntry));
+        }
+    }).catch(function (err) {
+        throw new Error('Could not get Directory Entry for ' + zimURL, err);
+    });
+}
+
+/**
+ * A function to attach the tooltip CSS for popovers (NB this does not attach the box itself, only the CSS)
+ * @param {Document} doc The document to which to attach the popover stylesheet
+ * @param {Boolean} dark An optional parameter to adjust the background colour for dark themes (generally not needed for inversion-based themes)
+ */
+function attachKiwixPopoverCss (doc, dark) {
+    const colour = dark ? '#darkgray' : '#black';
```

Comment:
```suggestion
    const colour = dark ? 'darkgray' : 'black';
```
Oops... Code wasn't tested because not used with current dark style, but will be used if/when we add `darkReader` to this Repo. See #908.

Reply:
Done.